crates/bevy_asset/src/path.rs

```diff
@@ -648,7 +648,12 @@ pub(crate) fn normalize_path(path: &Path) -> PathBuf {
         if elt == "." {
             // Skip
         } else if elt == ".." {
-            if !result_path.pop() {
+            if let Some(file_name) = result_path.file_name()
+                // Don't collapse .. with another ..
+                && file_name != ".."
+            {
+                assert!(result_path.pop());
```

Comment:
Does this mean bevy would crash given an invalid path that triggers this?

Reply:
No - this assert is literally just a sanity check. We already know we have a file name because we did `result_path.file_name()` and made sure it's `Some`. So we know for sure there's something in `result_path` to pop.